src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -52,6 +147,7 @@ internal void FunctionCompleted(FunctionStartedEvent completedEvent)
                     if (!instance.IsActive)
                     {
                         instance.StopEtwTaskAndRaiseFinishedEvent();
+                        instance.Dispose();
```

Comment:
My goal is to ensure that we flush events/logs whenever the app is going down. Need to investigate other paths and ensure we're doing our best.